server/channels/app/emoji.go

```diff
@@ -233,7 +233,21 @@ func (a *App) GetMultipleEmojiByName(c request.CTX, names []string) ([]*model.Em
 		return nil, model.NewAppError("GetMultipleEmojiByName", "api.emoji.disabled.app_error", nil, "", http.StatusForbidden)
 	}
 
-	emoji, err := a.Srv().Store().Emoji().GetMultipleByName(names)
+	// Filtering out system emojis
+	i := 0
+	for _, n := range names {
+		if _, ok := model.GetSystemEmojiId(n); !ok {
+			names[i] = n
+			i++
+		}
+	}
+	names = names[:i]
```

Comment:
Oh, in case you weren't aware, I just copy paste this from here: https://github.com/golang/go/wiki/slicetricks#filter-in-place. It's a great place with some neat code snippets.

Reply:
Yeah, I've seen that before, it's great! But still, every time I encounter this specific technique it's like an emotional roller-coaster for me until I recall what it's doing and why :joy: